pkgs/development/tools/build-managers/mill/default.nix

```diff
@@ -0,0 +1,26 @@
+{ stdenv, lib, fetchurl, makeWrapper, jre, gnused }:
+ stdenv.mkDerivation rec {
+  name = "mill-${version}";
+  version = "0.3.5";
+    src = fetchurl {
+     url = "https://github.com/lihaoyi/mill/releases/download/${version}/${version}";
+     sha256 = "19ka81f6vjr85gd8cadn0fv0i0qcdspx2skslfksklxdxs2gasf8";
+  };
+   propagatedBuildInputs = [ jre ] ;
+  buildInputs = [ makeWrapper gnused ] ;
+   phases = "installPhase";
+   installPhase = ''
```

Comment:
If you're overriding `installPhase` you should add `runHook preInstall` and `runHook postInstall` to your implementation.

Reply:
Do I need to do that also if we only have our own self defined `installPhase`? I tested it locally and it works, but I guess that is because they are undefined?

Reply:
Yeah, because it allows people to use `overrideAttrs` and insert their own things without having to override your whole `installPhase`.

Reply:
of course! hehe, didn't think about that! Nice :-)